experimental/traffic-portal/src/app/api/profile.service.ts

```diff
@@ -103,4 +124,64 @@ export class ProfileService extends APIService {
 		return this.delete<ResponseProfile>(`profiles/${id}`).toPromise();
 	}
 
+	public async getParameters(idOrName: number | string): Promise<ResponseParameter>;
+	public async getParameters(): Promise<Array<ResponseParameter>>;
+	/**
+	 * Retrieves Parameters from the API.
+	 *
+	 * @param idOrName Specify either the integral, unique identifier (number) of a specific Parameter to retrieve,
+	 * or its name (string).
+	 * @returns The requested Parameter(s).
+	 */
+	public async getParameters(idOrName?: number | string): Promise<Array<ResponseParameter> | ResponseParameter> {
+		const path = "parameters";
+		if (idOrName !== undefined) {
+			let params;
+			switch (typeof idOrName) {
+				case "number":
+					params = {id: idOrName};
+					break;
+				case "string":
+					params = {name: idOrName};
+			}
+			const r = await this.get<[ResponseParameter]>(path, undefined, params).toPromise();
+			if (r.length !== 1) {
+				throw new Error(`Traffic Ops responded with ${r.length} Types by identifier ${idOrName}`);
```

Comment:
The tests for `getParameters` only cover happy paths; this line - the case where more than one Parameter exists with the given name or ID - is not covered.

Reply:
Niot sure how I could make it return two params with the same ID?

Reply:
```typescript
const data = {
	response: [
		{ configFile: "test", id: 1, lastUpdated: new Date(), name: "test", secure: false, value: "test" },
		{ configFile: "quest", id: 1, lastUpdated: new Date(), name: "quest", secure: false, value: "quest" },
	]
};
req.flush(data);
```

Reply:
Ah, didn't realize `flush` sets the body of the response